src/utilities/common/incrementalResult.ts

```diff
@@ -28,6 +30,18 @@ export function isExecutionPatchResult<T>(
   );
 }
 
+// This function detects an Apollo payload result before it is transformed
+// into a FetchResult via HttpLink; it cannot detect an ApolloPayloadResult
+// once it leaves the link chain.
+export function isApolloPayloadResult(
+  value: unknown
+): value is ApolloPayloadResult {
+  return (
+    isNonNullObject(value) &&
+    ("payload" in value || ("done" in value && typeof value.done === "boolean"))
+  );
```

Comment:
```suggestion
  return isNonNullObject(value) && "payload" in value;
```
No need to worry about `done` anymore! 🥳 🎈

Reply:
Missed that last spot, thanks! 🎉